vectortile/src/main/scala/geotrellis/vectortile/package.scala

```diff
@@ -0,0 +1,89 @@
+  * Any classes in the ''internal.*'' submodules are unique to the machinery
+  * of VectorTile {de,en}coding, and can be safely ignored.
+  *
+  * ===Types===
+  *
+  * The central type is the [[VectorTile]] trait. At current, it is only
+  * backed by one backend, the default Protobuf implementation that follows the
+  * original spec. The accompanying type is
+  * [[geotrellis.vectortile.protobuf.ProtobufTile]], and its companion object can
+  * be used to actually construct VectorTiles from raw byte data:
+  *
+  * {{{
+  * import geotrellis.spark.SpatialKey
+  * import geotrellis.spark.tiling.LayoutDefinition
+  * import geotrellis.vectortile.VectorTile
+  * import geotrellis.vectortile.protobuf._
+  *
+  * val bytes: Array[Byte] = ...  // from some `.mvt` file
+  * val key: SpatialKey = ...  // preknown
+  * val layout: LayoutDefinition = ...  // preknown
+  *
+  * /* Decode Protobuf bytes. */
+  * val tile: VectorTile = ProtobufTile.fromBytes(bytes, key, layout)
+  *
+  * /* Encode a VectorTile back into bytes. */
+  * val encodedBytes: Array[Byte] = tile match {
+  *   case t: ProtobufTile => t.toBytes
+  *   case _ => ???  // Handle other backends or throw errors.
+  * }
+  *
+  * }}}
+  *
+  * The ''V*'' types form a small [[https://en.wikipedia.org/wiki/Sum_type sum type]]
+  * and are used to represent usually untyped Feature-level metadata. This metadata
+  * is equivalent to a JSON Object, where String keys index values of any type.
+  * A Scala Map requires more rigidity (for the better), and so we use
+  * [[geotrellis.vectortile.protobuf.Value]] to guarantee type safety.
+  *
+  * ==Implementation Assumptions==
+  * This particular implementation of the VectorTile spec makes the following
+  * assumptions:
+  *   - Geometries are implicitly encoded in ''some'' Coordinate Reference
+  *     system. That is, there is no such thing as a "projectionless" VectorTile.
+  *     When decoding a VectorTile, we must provide a Geotrellis [[LayoutDefinition]]
+  *     and a [[SpatialKey]]. The LayoutDefinition includes an Extent in some CRS,
+  *     which conceptually surrounds a grid of tiles on a map. The SpatialKey
+  *     represents the Tile's position in that grid.
+  *     With these, the grid coordinates stored in the VectorTile's Geometry are
+  *     shifted from their
+  *     original [0,4096] range to actual world coordinates in the Extent's CRS.
+  *   - The `id` field in VectorTile Features doesn't matter.
+  *   - If a VectorTile `geometry` list marked as `POINT` has only one pair
+  *     of coordinates, it will be decoded as a Geotrellis `Point`. If it has
+  *     more than one pair, it will be decoded as a `MultiPoint`. Likewise for
+  *     the `LINESTRING` and `POLYGON` types. A complaint has been made about
+  *     the spec regarding this, and future versions may include a difference
+  *     between single and multi geometries.
+  *
+  * @author cwoodbury@azavea.com
+  * @version 2.1
+  */
```

Comment:
There are great ScalaDocs, but I see there's no markdown docs...we've generally been putting this sort of doc in markdown instead of in ScalaDoc. I'm not sure how to avoid duplication between markdown and scaladocs, but I would want to see something for the markdown...would linking to the scaladocs work? That might be a bit weird.